### ISSUE TITLE
Can i input the parameter "gradient accumulation steps" in some way?

### ISSUE THREAD

**Issue by RaySteve312** [kijai/ComfyUI-FluxTrainer#34]
Can i input the parameter "gradient accumulation steps" in some way?
 I watched several training tutorials, they all mentioned this parameter."Gradient accumulation steps" can be used with "batch size"simply to save training time.
![屏幕截图 2024-09-03 162852](https://github.com/user-attachments/assets/b0e9dea3-175b-47ce-97a8-10765d3e38b5)

**Comment by kijai** [comment]
You can input it in the additional_args box:
![image](https://github.com/user-attachments/assets/5e5cdc1b-b366-4729-85c4-2876fdd347c6)

**Comment by Orenji-Tangerine** [comment]
> You can input it in the additional_args box: ![image](https://private-user-images.githubusercontent.com/40791699/364494486-5e5cdc1b-b366-4729-85c4-2876fdd347c6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1MjQ2MDAsIm5iZiI6MTcyNTUyNDMwMCwicGF0aCI6Ii80MDc5MTY5OS8zNjQ0OTQ0ODYtNWU1Y2RjMWItYjM2Ni00NzI5LTg1YzQtMjg3NmZkZDM0N2M2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA1VDA4MTgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmYjllYjllNDU3N2Y5YjBkYWE5YzA5NDYwZGRmN2EzNzkzNzhjNjYzMTI3N2NjNjdmZjI2OWE0NmE1MjQ0MjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4x1YIsCEVtK5GPmG8SkIgO9qus0f5amHh8aIgcI9QWM)

Hi Kijai, I notice your node has train_text_encoder,  may I know it is for what purpose? Is that for the next update?

**Comment by kijai** [comment]
> > You can input it in the additional_args box: ![image](https://private-user-images.githubusercontent.com/40791699/364494486-5e5cdc1b-b366-4729-85c4-2876fdd347c6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1MjQ2MDAsIm5iZiI6MTcyNTUyNDMwMCwicGF0aCI6Ii80MDc5MTY5OS8zNjQ0OTQ0ODYtNWU1Y2RjMWItYjM2Ni00NzI5LTg1YzQtMjg3NmZkZDM0N2M2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA1VDA4MTgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJmYjllYjllNDU3N2Y5YjBkYWE5YzA5NDYwZGRmN2EzNzkzNzhjNjYzMTI3N2NjNjdmZjI2OWE0NmE1MjQ0MjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4x1YIsCEVtK5GPmG8SkIgO9qus0f5amHh8aIgcI9QWM)
> 
> Hi Kijai, I notice your node has train_text_encoder,  may I know it is for what purpose? Is that for the next update?

It's currently under testing in another branch, it allows training a LoRA for the T5 as well.

**Comment by RaySteve312** [comment]
thanks for your work! you are my hero!

**Comment by RaySteve312** [comment]
> You can input it in the additional_args box: ![image](https://private-user-images.githubusercontent.com/40791699/364494486-5e5cdc1b-b366-4729-85c4-2876fdd347c6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2ODgwODIsIm5iZiI6MTcyNTY4Nzc4MiwicGF0aCI6Ii80MDc5MTY5OS8zNjQ0OTQ0ODYtNWU1Y2RjMWItYjM2Ni00NzI5LTg1YzQtMjg3NmZkZDM0N2M2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA3VDA1NDMwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5OWE3YzFkMDdlMzU0NmVlMDM2NzM3MmVkMTEwODkwMGRiZWQzMTY3YTdiYTY2MzE1NWExNTFlM2E0OTNhMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4-lbhHI5LeDTkNJ4ZoFxwGArSju2jQJplpGn0XsA824)
![屏幕截图 2024-09-07 134810](https://github.com/user-attachments/assets/36c0ae53-1218-4e7d-b97b-a440747de6b8)


I want to input 2 args at the same time. but it doesn't work. Is the input format wrong?

**Comment by kijai** [comment]
> 
> 
> > You can input it in the additional_args box: ![image](https://private-user-images.githubusercontent.com/40791699/364494486-5e5cdc1b-b366-4729-85c4-2876fdd347c6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2ODgwODIsIm5iZiI6MTcyNTY4Nzc4MiwicGF0aCI6Ii80MDc5MTY5OS8zNjQ0OTQ0ODYtNWU1Y2RjMWItYjM2Ni00NzI5LTg1YzQtMjg3NmZkZDM0N2M2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA3VDA1NDMwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5OWE3YzFkMDdlMzU0NmVlMDM2NzM3MmVkMTEwODkwMGRiZWQzMTY3YTdiYTY2MzE1NWExNTFlM2E0OTNhMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4-lbhHI5LeDTkNJ4ZoFxwGArSju2jQJplpGn0XsA824)
> ![屏幕截图 2024-09-07 134810](https://github.com/user-attachments/assets/36c0ae53-1218-4e7d-b97b-a440747de6b8)
> 
> 
> I want to input 2 args at the same time. but it doesn't work. When I input either arg of them, it works. 
> Is the input format wrong?

Yeah I think I had an error there, it's fixed in my current train_t5 branch which I can hopefully merge soon. It will be parsed like command line arguments would.

**Comment by RaySteve312** [comment]
> > > You can input it in the additional_args box: ![image](https://private-user-images.githubusercontent.com/40791699/364494486-5e5cdc1b-b366-4729-85c4-2876fdd347c6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2ODgwODIsIm5iZiI6MTcyNTY4Nzc4MiwicGF0aCI6Ii80MDc5MTY5OS8zNjQ0OTQ0ODYtNWU1Y2RjMWItYjM2Ni00NzI5LTg1YzQtMjg3NmZkZDM0N2M2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA3VDA1NDMwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5OWE3YzFkMDdlMzU0NmVlMDM2NzM3MmVkMTEwODkwMGRiZWQzMTY3YTdiYTY2MzE1NWExNTFlM2E0OTNhMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4-lbhHI5LeDTkNJ4ZoFxwGArSju2jQJplpGn0XsA824)
> > > ![屏幕截图 2024-09-07 134810](https://private-user-images.githubusercontent.com/134742529/365345905-36c0ae53-1218-4e7d-b97b-a440747de6b8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MDIwODIsIm5iZiI6MTcyNTcwMTc4MiwicGF0aCI6Ii8xMzQ3NDI1MjkvMzY1MzQ1OTA1LTM2YzBhZTUzLTEyMTgtNGU3ZC1iOTdiLWE0NDA3NDdkZTZiOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwN1QwOTM2MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNTVkZmNjMDZkNDVlYmFmMWM3ZmQwZGZkZWFlZWEzMGQ2NWM3MmZkMDgxOGYxZDM0OTFhODY0ZjMwZDg2ODNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.HSMTyUxsfeKDVqGp5XCxFGGOF3glu7Wk3Suw54yJa2g)
> > 
> > 
> > I want to input 2 args at the same time. but it doesn't work. When I input either arg of them, it works.
> > Is the input format wrong?
> 
> Yeah I think I had an error there, it's fixed in my current train_t5 branch which I can hopefully merge soon. It will be parsed like command line arguments would.

that would be great!🍻

**Comment by RaySteve312** [comment]
> > > You can input it in the additional_args box: ![image](https://private-user-images.githubusercontent.com/40791699/364494486-5e5cdc1b-b366-4729-85c4-2876fdd347c6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU2ODgwODIsIm5iZiI6MTcyNTY4Nzc4MiwicGF0aCI6Ii80MDc5MTY5OS8zNjQ0OTQ0ODYtNWU1Y2RjMWItYjM2Ni00NzI5LTg1YzQtMjg3NmZkZDM0N2M2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA3VDA1NDMwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5OWE3YzFkMDdlMzU0NmVlMDM2NzM3MmVkMTEwODkwMGRiZWQzMTY3YTdiYTY2MzE1NWExNTFlM2E0OTNhMDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4-lbhHI5LeDTkNJ4ZoFxwGArSju2jQJplpGn0XsA824)
> > > ![屏幕截图 2024-09-07 134810](https://private-user-images.githubusercontent.com/134742529/365345905-36c0ae53-1218-4e7d-b97b-a440747de6b8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MDQ0ODAsIm5iZiI6MTcyNTcwNDE4MCwicGF0aCI6Ii8xMzQ3NDI1MjkvMzY1MzQ1OTA1LTM2YzBhZTUzLTEyMTgtNGU3ZC1iOTdiLWE0NDA3NDdkZTZiOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwN1QxMDE2MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00Y2VmMTkzOWExN2NhMTY2YTA2MzUwODc3ODQ5Y2FmZGNlMzM1NWIwYzk2NzhiZGVmOTYwMjIyNTYzYTJjNWRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ubg769jbwYz1FEftZCA8qd-uQtvrh2m2LuWLned2Hto)
> > 
> > 
> > I want to input 2 args at the same time. but it doesn't work. When I input either arg of them, it works.
> > Is the input format wrong?
> 
> Yeah I think I had an error there, it's fixed in my current train_t5 branch which I can hopefully merge soon. It will be parsed like command line arguments would.

one more question, how to input reg images and repeat reg images, and set a weight? since regulariation plays an important role in training styles lora, I just read some introduction about it, and I find L2 in the node  Init Flux Training. It's soooo much to learn.
I 
![屏幕截图 2024-09-07 182438](https://github.com/user-attachments/assets/d1b17390-29e3-4e49-bbef-81899e675539)
